### PR TITLE
drivers/sensors: Add support to Renesas FS3000 Air Velocity Flow

### PR DESCRIPTION
**Pull request by acassis**
## Summary

This driver add uORB sensor support to Renesas FS3000 sensor.

## Impact

Users will be able to use FS3000 sensor on NuttX

## Testing
using US082-FS3000EVZ  module

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR summary does not adequately meet the NuttX requirements.  Here's why and how to fix it:

* **Insufficient Detail in Summary:**  The summary is too brief.  It mentions adding uORB support, but doesn't explain *why* this is necessary (e.g., was there no FS3000 support before? Was existing support inadequate?). It also lacks detail on *how* the change works (e.g., what files were modified, new files added, core logic implemented).  Mentioning the specific functional area (e.g., sensors/drivers) is also helpful.  Finally, link any related NuttX issues.

* **Incomplete Impact Assessment:** While it mentions user impact, it's very general.  You need to address *all* the impact points:
    * **Is new feature added? Is existing feature changed?** Yes, a new feature (FS3000 support) is added.
    * **Impact on build:**  Likely NO, but confirm. If any Kconfig changes are needed, specify them.
    * **Impact on hardware:** YES, specifically for Renesas platforms and boards using the FS3000. List the specific architectures/boards.
    * **Impact on documentation:** YES, documentation needs to be added to explain how to use the new driver.  State whether this documentation is included in the PR.
    * **Impact on security, compatibility:**  Probably NO for both, but explicitly state this and justify it briefly if there's any potential concern.

* **Missing Testing Details:**  "using US082-FS3000EVZ module" is not enough.  Provide the required details:
    * **Build Host:**  OS, CPU architecture, compiler version (e.g., Linux x86_64, GCC 12.2.0)
    * **Target:** Architecture, board, configuration (e.g., `armv7-m`, `stm32f4discovery`, `nsh`).  Be very specific.
    * **Logs:**  You *must* include "before" and "after" logs demonstrating the driver's functionality.  Show sensor data being read and published via uORB.  Without logs, reviewers cannot verify the change.



**Example of an improved summary:**

```
## Summary

This PR adds support for the Renesas FS3000 sensor by implementing a new driver and integrating it with the uORB sensor framework. Previously, the FS3000 was not supported in NuttX. This driver allows applications to access sensor data (e.g., [specify data types]) from the FS3000 via uORB topics.  The driver interacts with the FS3000 using [I2C/SPI/other interface] and publishes sensor readings to the `/sensor/fs3000` uORB topic.  Key changes include adding the driver code to `drivers/sensors/fs3000.c` and updating the configuration files to enable the driver.

Relates to [NuttX Issue # if applicable]
```

By providing more complete and specific information, you make it easier for reviewers to understand and approve your PR. Remember, a good PR description saves everyone time and leads to a faster integration process.

File: drivers/sensors/fs3000_uorb.c [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+ *
+ * Description:
+ *   Convert raw data to air velocity
+ *
+ * Return value:
+ *   Zero (OK) on success; a negated errno value on failure
+ ****************************************************************************/
+
+static int fs3000_get_velocity(FAR struct fs3000_dev_s *priv, uint8_t *data,
+                            FAR float *airvelocity)
```
Comment (xiaoxiang781216) [inline drivers/sensors/fs3000_uorb.c:149]:
```suggestion
                               FAR float *airvelocity)
```

File: drivers/sensors/fs3000_uorb.c [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: fs3000_register
+ *
+ * Description:
+ *   Register the FS3000 character device
+ *   @devno - The user-specified device number, starting from 0
```
Comment (xiaoxiang781216) [inline drivers/sensors/fs3000_uorb.c:238]: remove devno and correct the comment

File: drivers/sensors/fs3000_uorb.c [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+  FAR struct fs3000_dev_s *priv = (FAR struct fs3000_dev_s *)
+        ((uintptr_t)strtoul(argv[1], NULL, 16));
+  struct sensor_velocity velocity;
+  uint8_t data[5];
+  float airvelocity = 0.0;
+  int ret;
+
+  while (true)
+    {
+      ret = fs3000_get_data(priv, (uint8_t *) &data[0]);
```
Comment (xiaoxiang781216) [inline drivers/sensors/fs3000_uorb.c:204]:
```suggestion
      ret = fs3000_get_data(priv, data);
```

File: drivers/sensors/fs3000_uorb.c [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+      if (ret < 0)
+        {
+          snerr("ERROR: Failed to read data from sensor\n");
+          return ret;
+        }
+
+      ret = fs3000_get_velocity(priv, (uint8_t *)data, &airvelocity);
```
Comment (xiaoxiang781216) [inline drivers/sensors/fs3000_uorb.c:211]:
```suggestion
      ret = fs3000_get_velocity(priv, data, &airvelocity);
```

File: drivers/sensors/fs3000_uorb.c [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+      *airvelocity = -1.0f;
+      return -EINVAL;
+    }
+
+  value = (data[2] << 4) | (data[1] & 0xf) ;
+
+  /* Use the right correlation formula depending on sensor model */
+
+#if defined(CONFIG_SENSOR_FS3000_1005)
+  *airvelocity = (float) (0.00219 * value) - 1.17;
```
Comment (xiaoxiang781216) [inline drivers/sensors/fs3000_uorb.c:170]:
```suggestion
  *airvelocity = 0.00219f * value - 1.17f;
```

File: drivers/sensors/fs3000_uorb.c [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+#else
+  *airvelocity = (float) (0.00275 * value) - 1.1;
```
Comment (xiaoxiang781216) [inline drivers/sensors/fs3000_uorb.c:172]:
```suggestion
  *airvelocity = 0.00275f * value - 1.1f;
```

File: drivers/sensors/fs3000_uorb.c [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+/* This is struct is just for "aesthetics", not used */
+
+struct fs3000_sensor_s
+{
+  struct sensor_lowerhalf_s lower;  /* Common lower interface */
+};
+
+struct fs3000_dev_s
+{
+  struct fs3000_sensor_s dev;     /* Sensor private data */
```
Comment (xiaoxiang781216) [inline drivers/sensors/fs3000_uorb.c:63]:
```suggestion
  struct sensor_lowerhalf_s lower;     /* Sensor private data */
```
and remove fs3000_sensor_s

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**